### ISSUE TITLE
Julia crashing on Windows

### ISSUE THREAD

**Issue by lkapelevich** [JuliaOpt/MathOptInterface.jl#735]
I'm having issues with my Julia sessions crashing- this is a little tricky to describe because the crashing and error messages have been random. At least one way to trigger the issue is the following:

If I run a file with the contents below a couple of times, my session crashes.

```julia
import MathOptInterface
const MOI = MathOptInterface
MOIU = MOI.Utilities

MOIU.@model(SomeModelData,
    (),
    (MOI.EqualTo, MOI.GreaterThan, MOI.LessThan, MOI.Interval),
    (MOI.Reals, MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives,
        MOI.SecondOrderCone, MOI.RotatedSecondOrderCone,
        MOI.PositiveSemidefiniteConeTriangle,
        MOI.ExponentialCone, MOI.GeometricMeanCone, MOI.LogDetConeTriangle),
    (MOI.PowerCone,),
    (MOI.SingleVariable,),
    (MOI.ScalarAffineFunction,),
    (MOI.VectorOfVariables,),
    (MOI.VectorAffineFunction,),
    )

error()
```

The crashing happens either with no error or with an `EXCEPTION_ACCESS_VIOLATION` e.g. https://gist.github.com/lkapelevich/90b7c8d20ab0ef41f36b20c53ba2c11c. 

Version info:

```julia
julia> versioninfo()
Julia Version 1.3.0-DEV.220
Commit b4b46342ea (2019-05-14 16:15 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)

(Toy) pkg> st
Project Toy v0.1.0
    Status `C:\Users\lkape\.julia\dev\Toy\Project.toml`
  [b8f27783] MathOptInterface v0.8.4
```

**Comment by blegat** [comment]
It seems to be a Julia bug, we should open an issue in https://github.com/JuliaLang/julia

**Comment by odow** [comment]
Although, wow, you're running ahead of the pack on 1.3?

**Comment by lkapelevich** [comment]
A couple of things got merged after 1.2 that we need

**Comment by lkapelevich** [comment]
This doesn't happen anymore on the latest Julia nightly.
